translate/storage/pypo.py

```diff
@@ -757,20 +760,23 @@ class pofile(pocommon.pofile):
     """A .po file containing various units"""
     UnitClass = pounit
 
-    def parse(self, input):
+    def parse(self, input, duplicatestyle=None):
         """Parses the given file or file source string."""
-        if True:
-            if hasattr(input, 'name'):
-                self.filename = input.name
-            elif not getattr(self, 'filename', ''):
-                self.filename = ''
-            if isinstance(input, str):
-                input = BytesIO(input)
-            # clear units to get rid of automatically generated headers before parsing
-            self.units = []
-            poparser.parse_units(poparser.ParseState(input, pounit), self)
-
-    def removeduplicates(self, duplicatestyle="merge"):
+        if hasattr(input, 'name'):
+            self.filename = input.name
+        elif not getattr(self, 'filename', ''):
+            self.filename = ''
+        if isinstance(input, str):
+            input = BytesIO(input)
+        # clear units to get rid of automatically generated headers before parsing
+        self.units = []
+        poparser.parse_units(poparser.ParseState(input, pounit), self)
+
+        # duplicates are now removed by default unless duplicatestyle=allow
+        if duplicatestyle != "allow":
```

Comment:
@unho 

>@phlax What I was expecting of this PR is to keep previous behavior by default, but allowing > to discard duplicates in case it was explicitly specified when calling parse() (necessary for 
> Pootle)

we can do that but that would be more invasive for because when get_class(file)() gets called it would need to have the duplicatestyle param set - therefore all of the storagetypes would need to implement it or we would need to hardcode against po files